simulation/traffic_simulator/src/entity/entity_manager.cpp

```diff
@@ -254,13 +254,17 @@ auto EntityManager::getLongitudinalDistance(
 {
   if (!include_adjacent_lanelet) {
     auto forward_distance = hdmap_utils_ptr_->getLongitudinalDistance(
-      static_cast<LaneletPose>(from), static_cast<LaneletPose>(to));
+      from.getNonCanonicalizedLaneletPose(), to.getNonCanonicalizedLaneletPose());
```

Comment:
We believe that after the LaneletPose is Canonicalized, the Lanelet Pose before Canonicalization should no longer be needed.
The fix included in the /fix/longitudinal_distance branch fixes a bug that was caused by the Lanelet Pose before it was Canonicalized and allows the NonCanonicalizedLaneletPose to be used with the If you use NonCanonicalizedLaneletPose to calculate longitudinal distance, the bug will not be corrected.

Reply:
The reason, why I've decided to store the Lanelet Pose before Canonicalization and use it in the `hdmap_utils_ptr_->getLongitudinalDistance()` is that when the LaneletPose is Canonicalizated then [only first lanelet_id with the lowest array index is used as a candidate](https://github.com/tier4/scenario_simulator_v2/blob/fd4ff7a26613402134bf9737d6ef98e11d664b4b/simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp#L99). 
For a particular example, which is described at the 4. point in Description section in this PR, Canonicalization only to the first lanelet_id causes the scenario fails. 

Please, give me some time, I will try to find another solution for this issue.